spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverLoggingAutoConfiguration.java

```diff
@@ -38,6 +38,7 @@
  */
 @Configuration
 @ConditionalOnClass(TraceLoggingEnhancer.class)
+@ConditionalOnMissingBean(StackdriverTraceAutoConfiguration.class)
```

Comment:
@artembilan can you please let me know if you have any objections on this usage of `@ConditionalOnMissingBean` at the class level? Thanks!

Reply:
According `@ConditionalOnMissingBean` JavaDocs it is fully fine to have it on the `@Configuration` class:
```
 * The condition can only match the bean definitions that have been processed by the
 * application context so far and, as such, it is strongly recommended to use this
 * condition on auto-configuration classes only. If a candidate bean may be created by
 * another auto-configuration, make sure that the one using this condition runs after.
```

See for example a `XADataSourceAutoConfiguration`.

Reply:
However I see that you use the whole `StackdriverTraceAutoConfiguration`. Maybe that one provides some bean we can check here instead? Even if it is the same time as we produce here.
Don't forget to add also `@AutoConfigureAfter(StackdriverTraceAutoConfiguration.class)`

Reply:
@snicoll it might have been you who had issues with `@ConditionalOnMissingBean` at the class level.
If you have any issues with this, what do you suggest to make `StackdriverLoggingAutoConfiguration` only kick in if `StackdriverTraceAutoConfiguration` hasn't?

Reply:
> it might have been you who had issues with @ConditionalOnMissingBean at the class level.

As long as you understand what that means, i am cool with it (it's being applied on every `@Bean` methods.